opencga-master/src/main/java/org/opencb/opencga/master/monitor/daemons/ExecutionDaemon.java

```diff
@@ -324,22 +332,46 @@ protected void checkJobs() throws CatalogException {
             }
             logger.info("----- EXECUTION DAEMON  ----- Organization={} --> pending={}, queued={}, running={}", organizationId, pendingJobs,
                     queuedJobs, runningJobs);
+            totalPendingJobs += pendingJobs;
+            totalQueuedJobs += queuedJobs;
+            totalRunningJobs += runningJobs;
         }
 
-        /*
-        PENDING JOBS
-        */
-        checkPendingJobs(organizationIds);
+        if (totalQueuedJobs == 0) {
+            // If there are no queued jobs, we can queue new jobs
+            List<Job> pendingJobs = new LinkedList<>();
+            List<Job> runningJobs = new LinkedList<>();
 
-        /*
-        QUEUED JOBS
-        */
-        checkQueuedJobs(organizationIds);
+            for (String organizationId : organizationIds) {
+                try (DBIterator<Job> iterator = jobManager.iteratorInOrganization(organizationId, pendingJobsQuery, queryOptions, token)) {
+                    while (iterator.hasNext()) {
+                        pendingJobs.add(iterator.next());
+                    }
+                } catch (Exception e) {
+                    logger.error("{}", e.getMessage(), e);
+                }
 
-        /*
-        RUNNING JOBS
-        */
-        checkRunningJobs(organizationIds);
+                try (DBIterator<Job> iterator = jobManager.iteratorInOrganization(organizationId, runningJobsQuery, queryOptions, token)) {
+                    while (iterator.hasNext()) {
+                        runningJobs.add(iterator.next());
+                    }
+                } catch (Exception e) {
+                    logger.error("{}", e.getMessage(), e);
+                }
+            }
+
+            jobScheduler.addJobs(pendingJobs, Collections.emptyList(), runningJobs);
```

Comment:
This section should be moved to the existing method "checkPendingJobs"

Reply:
Done